doc/conf.py

```diff
@@ -690,7 +691,11 @@ def append_attr_meth_examples(app, what, name, obj, options, lines):
 
 .. minigallery:: {1}
 
-""".format(name.split(".")[-1], name).split("\n")
+""".format(
+                name.split(".")[-1], name
+            ).split(
+                "\n"
+            )
```

Comment:
This... cannot be right :)

Can you modify `ruff format mne` in `.pre-commit-config.yaml` to have
```
        files: ^mne/|^doc/
```
and see if it fixes this when you run `pre-commit run --all`?

Reply:
ohh.... weird!! I didn't realize that my auto-formatter did this... Will do.

Reply:
FYI I made your suggested change to `.pre-commit-config.yaml` but it didn't seem to affect anything. The problem seems to be with  the `format on save` option in my VS code settings (for my black formatter). Black automatically applies that weird formatting whenever I open `doc/conf.py` and save it. So my workaround was just to turn off the format on save option and fix the formatting manually...

Not sure if others can replicate that behaviour. But let me know if you want me to revert the change to `.pre-commit-config.yaml` since it didn't seem to help?

Reply:
our codebase is formatted with Ruff nowadays, not Black.  So changing VSCode to use Ruff as its "format-on-save" provider might help?

Reply:
Thanks @drammock that did the trick!

Reply:
> The problem seems to be with the format on save option in my VS code settings (for my black formatter). 

You can leave this on globally and save an individual file without running the auto formatter by selecting "Save without formatting" from the command palette